unlock-app/src/components/interface/locks/Settings/forms/UpdateReferralFee.tsx

```diff
@@ -16,82 +17,90 @@ interface UpdateReferralFeeProps {
 
 interface FormProps {
   referralFeePercentage: number
+  referralAddress: string
 }
 
-const ZERO = ethers.constants.AddressZero
+interface ReferrerFee {
+  id: string
+  referrer: string
+  fee: string
+}
 
 export const UpdateReferralFee = ({
   lockAddress,
   network,
   isManager,
   disabled,
 }: UpdateReferralFeeProps) => {
-  const [referralFee, setReferralFee] = useState(false)
+  const [isReferralFeeEnabled, setIsReferralFeeEnabled] = useState(false)
+  const [isReferralAddressEnabled, setIsReferralAddressEnabled] =
+    useState(false)
   const web3Service = useWeb3Service()
   const { getWalletService } = useAuth()
 
   const {
     register,
     handleSubmit,
-    setValue,
+    reset,
     formState: { errors },
-  } = useForm<FormProps>()
+  } = useForm<FormProps>({
+    defaultValues: { referralFeePercentage: 0, referralAddress: '' },
+  })
 
   const setReferrerFee = async (fields: FormProps) => {
     const walletService = await getWalletService(network)
     await walletService.setReferrerFee({
       lockAddress,
-      address: ZERO,
+      address: fields?.referralAddress,
       feeBasisPoint: fields?.referralFeePercentage * 100,
     })
   }
 
-  const getReferrerFees = async () => {
-    return await web3Service.referrerFees({
+  const getReferrerFees = async (): Promise<ReferrerFee[]> => {
+    return web3Service.referrerFees({
       lockAddress,
       network,
-      address: ZERO,
     })
   }
 
   const setReferrerFeeMutation = useMutation(setReferrerFee)
 
   const onSubmit = async (fields: FormProps) => {
-    const setReferrerFeePromise = setReferrerFeeMutation.mutateAsync(fields)
+    const setReferrerFeePromise = setReferrerFeeMutation.mutateAsync({
+      ...fields,
+      referralAddress: ethers.utils.getAddress(
+        fields.referralAddress.toLowerCase()
+      ),
+    })
 
     await ToastHelper.promise(setReferrerFeePromise, {
-      loading: 'Updating referrer fee',
+      loading: 'Setting new referrer',
       error: 'Failed to update the values, please try again.',
-      success: 'Referrer fee updated.',
+      success: 'New referrer set',
     })
+
+    reset()
   }
 
-  const { isLoading, data: referralFeePercentage } = useQuery(
+  const { isLoading, data: referralFeesData } = useQuery(
     ['getReferrerFees', lockAddress, network, setReferrerFeeMutation.isSuccess],
     async () => getReferrerFees()
   )
 
-  useEffect(() => {
-    setValue('referralFeePercentage', (referralFeePercentage ?? 0) / 100)
-    setReferralFee((referralFeePercentage ?? 0) > 0)
-  }, [referralFeePercentage])
+  const referralFees = referralFeesData || []
 
-  const disabledInput =
+  const isDisabledReferrerInput =
```

Comment:
Ensure that variable naming is clear for other developers to understand. It should always start with "is", "has", or "should".